### PR TITLE
Add StagehandTesting to SPM package, update CI

### PR DESCRIPTION
**Pull request by luispadron**
Exposes StagehandTesting via the `Package.swift`. Cleans up some more Bazel references and updates CI to test SPM via `xcodebuild` which allows building the package targets but with the correct platforms. Using `swift build` it was trying to link the macOS version of XCTest even though i was specifying the iOS sdk 🤷🏼

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -28,11 +28,13 @@ jobs:
       fail-fast: false
     steps:
       - name: Checkout Repo
-        uses: actions/checkout@v2
-      - name: Prepare Simulator Runtimes
-        run: Scripts/github/prepare-simulators.sh ${{ matrix.platform }}
```
Comment (dfed) [inline .github/workflows/ci.yml:33]: good to see this go away! Haven't needed this for a few release cycles now. We should probably delete the script from git as well though.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -28,11 +28,13 @@ jobs:
...
+        uses: actions/checkout@v4
```
Comment (dfed) [inline .github/workflows/ci.yml:27]: :pray:

File: Package.swift [hunk 3]
```diff
@@ -22,19 +22,39 @@ let package = Package(
 	name: "Stagehand",
 	platforms: [
 		.iOS(.v12),
+		.macOS(.v11),
```
Comment (dfed) [inline Package.swift:25]: Why is this necessary? A bit of a surprising change given that Stagehand does not support macOS (per the Podspec).

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -28,11 +28,13 @@ jobs:
...
       - name: Build
         run: |
-          swift build \
-            --sdk "$(xcode-select -p)/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS.sdk" \
-            --triple "arm64-apple-ios13.0"
+          # select minimum supported Xcode version
+          sudo xcode-select -s /Applications/Xcode_14.3.1.app
+          # test on the minimum supported version of the github actions runner
```
Comment (dfed) [inline .github/workflows/ci.yml]: nit: the next line isn't actually running tests – it's just building.

**Review by dfed — COMMENTED** [review]
Per [my comment on #75](https://github.com/cashapp/stagehand/pull/75#pullrequestreview-2326764213), it'd be good to re-add unit tests to CI. Could maybe be part of this PR?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (NickEntin) [reply .github/workflows/ci.yml:33]: I think we needed this because the iOS versions trailed behind a bit. Agreed we should update to the snapshots to latest iOS version too though.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (NickEntin) [reply .github/workflows/ci.yml]: +1, and I think also only building the framework, not the demo app

File: Package.swift (hunk 3 above)
Comment (luispadron) [reply Package.swift:25]: It's required because of the snapshot-testing dependency requiring a newer macOS version than the SPM default (macOS 10.15). In SPM we can't actually say "we don't support this platform", we can only list which platform minimum we support, so this increases it to allow us to pull in swift-snapshot-testing

File: .github/workflows/ci.yml (hunk 4 above)
Comment (luispadron) [reply .github/workflows/ci.yml]: My change doesn't change what it builds, I'll update the comment though since we aren't running tests as the SPM package doesn't define any

File: .github/workflows/ci.yml (hunk 1 above)
Comment (luispadron) [reply .github/workflows/ci.yml:33]: I'll push an update to delete the script, afaik if we pass the destination to Xcode build it should prepare the right sim

**Comment by luispadron** [comment]
> Per [my comment on #75](https://github.com/cashapp/stagehand/pull/75#pullrequestreview-2326764213), it'd be good to re-add unit tests to CI. Could maybe be part of this PR?

Ah my bad, yes I can look at getting tests added via the SPM workflow / pod lib lint.

File: Package.swift (hunk 3 above)
Comment (dfed) [reply Package.swift:25]: Fascinating! Thanks for the explainer. I'd suggest adding a comment here so that this can be deleted in the future when minimum supported versions change (or when SPM gets better at this, which... hopefully sooner rather than later)

**Comment by luispadron** [comment]
@dfed see #77 as to why re-adding example CI is out-of-scope for this PR. I hope ill have time to get the examples updated but right now GitHub actions do not support the iOS versions / Xcode versions the example tests require. I'm focused on adding better support for SPM as that's what were moving to internally.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -8,31 +8,27 @@ on:
...
-      - name: Bundle Install
-        run: bundle install --gemfile=Example/Gemfile
-      - name: Pod Install
-        run: bundle exec --gemfile=Example/Gemfile pod install --project-directory=Example
-      - name: Lint Stagehand Podspec
-        run: bundle exec --gemfile=Example/Gemfile pod lib lint --verbose --fail-fast Stagehand.podspec
-      - name: Lint StagehandTesting Podspec
-        run: bundle exec --gemfile=Example/Gemfile pod lib lint --verbose --fail-fast --include-podspecs=Stagehand.podspec StagehandTesting.podspec
+      - name: Lint podspecs
+        run: pod lib lint --verbose --fail-fast --include-podspecs=Stagehand.podspec StagehandTesting.podspec
```
Comment (dfed) [inline .github/workflows/ci.yml]: why do we not want `bundle exec` here? We should always ensure we're using the same cocoapods version to lint as the one we're going to publish with.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -8,31 +8,27 @@ on:
...
-        uses: actions/checkout@v2
-      - name: Prepare Simulator Runtimes
-        run: Scripts/github/prepare-simulators.sh ${{ matrix.platform }}
+        uses: actions/checkout@v4
       - name: Build
         run: |
-          swift build \
-            --sdk "$(xcode-select -p)/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS.sdk" \
-            --triple "arm64-apple-ios13.0"
+          # select minimum supported Xcode version
```
Comment (dfed) [inline .github/workflows/ci.yml:30]: future-looking comment:

we have [some prior art](https://github.com/square/Valet/pull/308#issuecomment-2071255780) for dropping Xcode version support in minor version bumps. We also do not state the minimum required Xcode version in this project's README 😅 so there's no real contract to break. (maybe we should fix that last part, but... that also doesn't need to be today)

File: Package.swift [hunk 7]
```diff
@@ -1,4 +1,4 @@
-// swift-tools-version:5.0.1
+// swift-tools-version:5.8
```
Comment (dfed) [inline Package.swift:1]: 🎉

File: Package.swift (hunk 3 above)
Comment (dfed) [reply Package.swift:25]: would still love a comment here 😅

File: Package.swift [hunk 8]
```diff
@@ -22,19 +22,39 @@ let package = Package(
...
 	],
 	products: [
 		.library(
 			name: "Stagehand",
 			targets: ["Stagehand"]
 		),
+		.library(
+			name: "StagehandTesting",
+			targets: ["StagehandTesting"]
+		),
```
Comment (dfed) [inline Package.swift:35]: love that this is getting exposed via SPM now!

File: Package.swift [hunk 9]
```diff
@@ -22,19 +22,39 @@ let package = Package(
...
+		.package(
+            url: "https://github.com/pointfreeco/swift-snapshot-testing.git",
+            .upToNextMajor(from: "1.8.0")
+        ),
 	],
 	targets: [
 		.target(
 			name: "Stagehand",
 			dependencies: [],
 			path: "Sources/Stagehand"
```
Comment (dfed) [inline Package.swift]: far from blocking, but I don't think you need this:
```suggestion
```

File: Package.swift [hunk 10]
```diff
@@ -22,19 +22,39 @@ let package = Package(
...
 		),
+		.target(
+			name: "StagehandTesting",
+			dependencies: [
+				"Stagehand",
+				.product(name: "SnapshotTesting", package: "swift-snapshot-testing"),
+			],
+			path: "Sources/StagehandTesting",
```
Comment (dfed) [inline Package.swift]: similar to the above, I don't think you need this:
```suggestion
```

**Review by dfed — COMMENTED** [review]
Change here generally makes sense to me, but getting CI's pod linting in sync with the Cocoapods version in Gemfile.lock is blocking imo.

File: .github/workflows/ci.yml (hunk 6 above)
Comment (luispadron) [reply .github/workflows/ci.yml:30]: The comment is referring to the minimum Xcode version on `macos-latest` so we can target the earliest iOS version on those workers, but Stagehand still supports iOS 12 and that hasn't changed. Updating the Xcode version used in CI for builds/tests shouldn't effect semver afaik

File: .github/workflows/ci.yml (hunk 5 above)
Comment (luispadron) [reply .github/workflows/ci.yml]: I had removed that because the root of the repo doesnt have a `Gemfile` and it was referring to `Example` here which i was going to address build & testing in #77. I added back the `bundle exec` until we add a root `Gemfile`

**Review by jszumski — APPROVED** [review]
